docs/caduceus.yaml

```diff
@@ -0,0 +1,235 @@
+        dest:
+          type: string
+          description: The name of the target device of the event.
+          example: "event:device-status/mac:ffffffffdae4/online"
+        content_type:
+          type: string
+          description: The media type of the payload.
+          example: "application/json"
+        partner_ids:
+          type: array
+          items:
+            type: string
+          description: The list of partner ids the message is meant to target.
+          example: ["example_partner", ".*"]
+        headers:
+          type: array
+          items:
+            type: string
+          description: The headers associated with the payload.
+        metadata:
+          type: object  # map[string]string
+          additionalProperties:
+            type: string
+          description: 
+            The map of name/value pairs used by consumers of WRP messages for 
+            filtering & other purposes.
+          example: {"/boot-time": "1542834188","/last-reconnect-reason": "spanish inquisition" }
+        payload:
+          type: string
+          format: binary
+          description: The bin format packed data.
+        session_id:
+          type: string
+          description: A unique ID for the device's connection session with XMiDT.
+          example: "1aaIkcz1cak0wXzIHsXkV8Uak1V"
+    Webhook:
+      description: The information needed to register a webhook with XMiDT.
+      type: object
+      required:
+      - events
+      properties:
+        registered_from_address:
+          type: string
+          description: The subscription request origin HTTP Address.
+          example: http://127.0.0.1/fakeurl/registerer
+        config:
+          type: object
+          description: Values related to event delivery.
+          properties:
+            url:
+              type: string
+              description: The HTTP URL to deliver messages to.
+              example: http://127.0.0.1/fakeurl/events
+            content_type:
+              type: string
+              description: 
+                The type of messages desired.  If "application/msgpack" is
+                specific, the full wrp is sent to the webhook.  If not, only the
+                payload is sent.
+              example: "application/msgpack"
+            secret:
+              type: string
+              description: The string value for the SHA1 HMAC.
+              example: "abcdefghijklmn"
+            alt_urls:
+              type: array
+              items:
+                type: string
+              description:
+                 The list of explicit URLs that should be round robin through on
+                 failure cases to the main URL.
+              example: []
+        failure_url:
+          type: string
+          description: 
+            The URL used to notify subscribers when they've been cut off due to 
+            event overflow.
+          example: http://127.0.0.1/fakeurl/failure
```

Comment:
Change to use https://failure-listener.example.com since localhost doesn't work.